src/linksys-addfwhdr.c

```diff
@@ -0,0 +1,191 @@
+/*
+	Linksys e8350 v1 firmware header generator
+*/
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <stdarg.h>
+#include <getopt.h>
+#include <time.h>
+#include <sys/time.h>
+#include "cyg_crc.h"
+#define AC2350          20
+
+#define CYBERTAN_VERSION        "v1.0.03"
+#define SERIAL_NUMBER   "003"
+#define MINOR_VERSION   ""
+#define BUILD_KEYWORD   " B"
+#define BUILD_NUMBER    SERIAL_NUMBER
+#define BETA_VERSION    " "
+#define CYBERTAN_UBOOT_VERSION      "v1.0"
+
+//add for AC2350 F/W header
+#define FWHDR_MAGIC_STR "CHDR"
+#define FWHDR_MAGIC 0X52444843
+struct cbt_fw_header
+{
+	unsigned int magic;             /* "CHDR" */
+	unsigned int len;               /* Length of file including header */
+	unsigned int crc32;             /* 32-bit CRC */
+	unsigned int res;
+};
+
+#define MAX_BUF	1024
+#define CRC32_INIT_VALUE 0xffffffff	/* Initial CRC32 checksum value */
+
+#ifndef TYPEDEF_UINT8
+typedef unsigned char   uint8;
+#endif
+
+#ifndef TYPEDEF_UINT16
+typedef unsigned short  uint16;
+#endif
+
+#ifndef TYPEDEF_UINT32
+typedef unsigned int    uint32;
+#endif
+
+typedef uint32_t cyg_uint32;
+typedef uint16_t cyg_uint16;
```

Comment:
Do we really needs this?

Reply:
> Do we really needs this?

Relics of the past. 
This whole code is based on what GPLed code firmware was doing with a firmware. I will try removing stuff gradually and see at what point it will break.

Reply:
Removed